src/workerd/api/streams/compression.c++

```diff
@@ -24,7 +24,8 @@ public:
     kj::ArrayPtr<const byte> buffer;
   };
 
-  explicit Context(Mode mode, kj::StringPtr format) : mode(mode) {
+  explicit Context(Mode mode, kj::StringPtr format, bool strictCompression) :
```

Comment:
Nit: generally not a fan of bool args like this. I'd prefer it to be an enum class like:

```
enum class ContextFlags {
  NONE,
  STRICT
};

explicit Context(Mode mode, kj::StringPtr format, ContextFlags flags);
```

This makes the call sites more descriptive (e.g. `ContextFlags::STRICT` as opposed to just `true`